scapy/volatile.py

```diff
@@ -783,6 +788,130 @@ def _fix(self):
         r = random.choice(self._pool)
         return r._fix()
 
+
+class RandUUID(RandField):
+    """Generates a random UUID.
+
+    By default, this generates a RFC 4122 version 4 UUID (totally random).
+
+    See Python's ``uuid`` module documentation for more information.
+
+    Args:
+        template (optional): A template to build the UUID from. Not valid with
+                             any other option.
+        node (optional): A 48-bit Host ID. Only valid for version 1 (where it
+                         is optional).
+        clock_seq (optional): An integer of up to 14-bits for the sequence
+                              number. Only valid for version 1 (where it is
+                              optional).
+        namespace: A namespace identifier, which is also a UUID. Required for
+                   versions 3 and 5, must be omitted otherwise.
+        name: string, required for versions 3 and 5, must be omitted otherwise.
+        version: Version of UUID to use (1, 3, 4 or 5). If omitted, attempts to
+                 guess which version to generate, defaulting to version 4
+                 (totally random).
+
+    Raises:
+        ValueError: on invalid constructor arguments
+    """
+    # This was originally scapy.contrib.dce_rpc.RandUUID.
+
+    _BASE = "([0-9a-f]{{{0}}}|\\*|[0-9a-f]{{{0}}}:[0-9a-f]{{{0}}})"
+    _REG = re.compile(
+        r"^{0}-?{1}-?{1}-?{2}{2}-?{2}{2}{2}{2}{2}{2}$".format(
+            _BASE.format(8), _BASE.format(4), _BASE.format(2)
+        ),
+        re.I
+    )
+    VERSIONS = (1, 3, 4, 5)
```

Comment:
I'd rather have a list here:
```suggestion
    VERSIONS = [1, 3, 4, 5]
```

Reply:
I explicitly used a `tuple` here because this shouldn't be modified at runtime -- it would require other code changes to support it (by subclassing it). What advantage is there to using a `list` instead?

Reply:
This is more a convention than a technical reason. See [Raymond Hettinger's tip](https://twitter.com/raymondh/status/324664257004322817) for example.